bob/src/cleaner.rs

```diff
@@ -52,19 +101,7 @@ impl Cleaner {
                 let hard = hard.unwrap_or(10);
                 backend.close_unneeded_active_blobs(soft, hard).await;
             }
-            if let Some(limit) = index_memory_limit {
-                let mut memory = backend.index_memory().await;
-                info!("Memory before closing old active blobs: {:?}", memory);
-                while memory > limit {
-                    if let Some(freed) = backend.close_oldest_active_blob().await {
-                        memory = memory - freed;
-                        debug!("closed index, freeing {:?} bytes", freed);
-                    } else {
-                        break;
-                    }
-                }
-                info!("Memory after closing old active blobs: {:?}", memory);
-            }
+            cleaner.request_index_cleanup().await;
```

Comment:
It is better to execute the code of clean-up here directly, and its execution should be synced with `fast_cleaner_task` with `Mutex`. And obviously, `close_unneeded_active_blobs` should also be protected by that mutex